fixtures/asset-config/auto-trailing-slash.test.ts

```diff
@@ -0,0 +1,226 @@
+import { SELF } from "cloudflare:test";
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
+import Worker from "../../packages/workers-shared/asset-worker/src/index";
+import { getAssetWithMetadataFromKV } from "../../packages/workers-shared/asset-worker/src/utils/kv";
+import type { AssetMetadata } from "../../packages/workers-shared/asset-worker/src/utils/kv";
+
+const IncomingRequest = Request<unknown, IncomingRequestCfProperties>;
+
+vi.mock("../../packages/workers-shared/asset-worker/src/utils/kv.ts");
+const existsMock = (fileList: Set<string>) => {
+	vi.spyOn(Worker.prototype, "exists").mockImplementation(
+		async (pathname: string) => {
+			if (fileList.has(pathname)) {
+				return pathname;
+			}
+		}
+	);
+};
+const BASE_URL = "http://example.com";
+
+type TestCase = {
+	title: string;
+	files: string[];
+	requestPath: string;
+	matchedFile?: string;
+	finalPath?: string;
+	// redirected?: boolean;
+};
+
+const testCases: TestCase[] = [
+	{
+		title: "/ -> 200 with /index.html",
+		files: ["/index.html"],
+		requestPath: "/index.html",
+		matchedFile: "/index.html",
+		finalPath: "/",
+	},
+	{
+		title: "/index -> / 307 (with /index.html)",
+		files: ["/index.html"],
+		requestPath: "/index",
+		matchedFile: "/index.html",
+		finalPath: "/",
+	},
+	{
+		title: "/index.html -> / 307 (with /index.html)",
+		files: ["/index.html"],
+		requestPath: "/index.html",
+		matchedFile: "/index.html",
+		finalPath: "/",
+	},
+	{
+		title: "/both -> 200 (with /both.html)",
+		files: ["/both.html", "/both/index.html"],
+		requestPath: "/both",
+		matchedFile: "/both.html",
+		finalPath: "/both",
+	},
+	{
+		title: "/both.html -> /both 307 (with /both.html)",
+		files: ["/both.html", "/both/index.html"],
+		requestPath: "/both",
+		matchedFile: "/both.html",
+		finalPath: "/both",
+	},
+	{
+		title: "/both/ -> 200 (with /both/index.html)",
+		files: ["/both.html", "/both/index.html"],
+		requestPath: "/both/",
+		matchedFile: "/both/index.html",
+		finalPath: "/both/",
+	},
+	{
+		title: "/both/index.html -> /both/ 307 (with /both/index.html)",
+		files: ["/both.html", "/both/index.html"],
+		requestPath: "/both/index.html",
+		matchedFile: "/both/index.html",
+		finalPath: "/both/",
+	},
+	{
+		title: "/both/index -> /both/ 307 (with /both/index.html)",
+		files: ["/both.html", "/both/index.html"],
+		requestPath: "/both/index",
+		matchedFile: "/both/index.html",
+		finalPath: "/both/",
+	},
+	{
+		title: "/file/ -> /file 307 (with file.html)",
+		files: ["/file.html"],
+		requestPath: "/file/",
+		matchedFile: "/file.html",
+		finalPath: "/file",
+	},
+	{
+		title: "/file/index -> /file 307 (with file.html)",
+		files: ["/file.html"],
+		requestPath: "/file/index",
+		matchedFile: "/file.html",
+		finalPath: "/file",
+	},
+	{
+		title: "/file/index.html -> /file 307 (with file.html)",
```

Comment:
I think I would expect this to 404?

Reply:
discussed: definitely very greedy matching, but consistent (in terms of what options are tried regardless of what files exist, and what Pages currently does)